src/main/java/core/training/TrainingWeekManager.java

```diff
@@ -190,66 +143,27 @@ private Instant getNextTrainingDate() {
 				}
 			}
 
-			zdtCurrDate = zdtCurrDate.plus(7, ChronoUnit.DAYS);
-			currDate = zdtCurrDate.toInstant();
+			currDate = currDate.plus(7, ChronoUnit.DAYS);
 		}
 
 		return trainings;
 	}
 
+	private int weekSinceOrigin(HODateTime.HTWeek htWeek) {
+		return htWeek.season*16+htWeek.week;
```

Comment:
This feels like this belongs in `HODateTime`.

Reply:
done